### ISSUE TITLE
findelement by Xpath is way too slower than findElementByName

### ISSUE THREAD

**Issue by vaib-ag** [Microsoft/WinAppDriver#638]
HI,
I am trying to find and attach to a window from desktop session.
When i use findElementByName the window is found within 1 second.

However while using xPath it takes approx 19 seconds. (After updating to v1.1.1 it takes approx 14-15 seconds.

Reason for using xPath is that name is not constant and hence using contains() of xpath.

**Comment by samchen63** [comment]
Yes, I have same issue. As in my testing application, there is not unique attribute to identify the element, so I have to use xpath. However it takes about 10+ seconds to find the element.

**Comment by JunlinZhu** [comment]
Using absolute xpath generated by UIRecorder should be much faster.

**Comment by samchen63** [comment]
> Using absolute xpath generated by UIRecorder should be much faster.

Thanks. Comparing with ID, using absolute xpath is still slow, but it is much faster than relative xpath.

**Comment by ankurgup510** [comment]
Hi, any update on this issue coz the problem still exists i.e findelementByXpath works very slow comparing others
